### ISSUE TITLE
Blob/Buffer confusion on iOS with JSPM, WebSQL SQLite adapter

### ISSUE THREAD

**Issue by WorldMaker** [pouchdb/pouchdb#6008]
### Issue
Attempting to save an attachment to a PouchDB backed by the Cordova SQLite 2 plugin and and getting an error `item.copy` is undefined deep in `Buffer.concat` in JSPM's Buffer polyfill for the browser. All of my attachment code is Blob oriented, so it seems like PouchDB may be confusing a Blob for a Buffer. First guess is of course simply the fact that JSPM has a polyfill for Node's Buffer available and require-able may cause this sort of track jump. Every other platform is using IndexedDB just fine, but we are trying to use Cordova WebSQL SQLite plugin on iOS to avoid some performance issues with attachment handling in iOS WebSQL...

### Info
- Environment: Cordova iOS UIWebView, JSPM/SystemJS
- Platform: iOS
- Adapter: Cordova WebSQL SQLite (Cordova SQLite Plugin 2)

### Reproduce
JSPM install the PouchDB Cordova SQLite adapter plugin and attempt to save an attachment. (I can try to build a minimal repro repo, I think, but this is the end of a long debugging day already (sigh, iOS), so expect one later if it is a big need.)

**Comment by nolanlawson** [comment]
hmm if I had to guess, I'd say the problem is [our jspm config](https://github.com/pouchdb/pouchdb/blob/9abef3a162f9a117497aaca9c3273d20cf7bb23c/packages/node_modules/pouchdb/package.json#L22-L24) which forces use of the browser bundle of PouchDB, whereas for pouchdb-adapter-cordova-websql it probably pulls in pouchdb-adapter-websql-core without preferring the browser version, therefore mixing the Node version and the browser version.

It seems you can force JSPM to honor the `"browser"` field by using e.g. `require('npm:packagename')`: https://github.com/jspm/jspm-cli/issues/1675 Could you give this a try and see if it fixes the issue?

**Comment by WorldMaker** [comment]
Ah, ok, I think I see. `pouchdb-binary-utils` could use a `jspm.main` override as well (at least until JSPM gets better browserify versus node support). I'm going to try to force the override:

```bash
$ jspm install npm:pouchdb-binary-utils@6.0.7 -o "{main:'lib/index-browser.js'}"
```

I'll let you know if iOS testing shows this works, but the overridden install files look correct.

**Comment by WorldMaker** [comment]
Looks like I need to override `pouchdb-md5` too:

```bash
$ jspm install npm:pouchdb-md5@6.0.7 -o "{main:'lib/index-browser.js'}"
```

**Comment by WorldMaker** [comment]
With those two overrides in place, things seem to work. (Even if the iOS performance still leaves something to be desired.)

**Comment by nolanlawson** [comment]
Performance issues probably has to do with the fact that asyncstorage is a simple key-value store. SQLite would undoubtedly be faster for most use cases.

**Comment by nolanlawson** [comment]
FWIW SQLite+PouchDB should be usable in ReactNative: https://twitter.com/nolanlawson/status/727281302332829696

**Comment by WorldMaker** [comment]
The performance issues are primarily attachment related. The sample test data we've been using for performance tests includes image attachments and UI that displays those image attachments quite prominently. From testing it seems like on iOS (UIWebView) performance is roughly the best with PouchDB+CBLite (version bundled with Cordova plugin, which I think it's out of date), then PouchDB+SQLite. In several ways I've discovered my sample images were more of a worst case stress test than I intended, but there's still some concern that it will be easily recreated (worse) in real world testing. I will be seeing what I can to mitigate some of it, but am definitely worried about image attachment performance on iOS short term, but even overall long term, in the age of crazy high megapixel cameras on mobile devices.
